source/adios2/toolkit/remote/remote_server.cpp

```diff
@@ -619,9 +619,14 @@ int main(int argc, char **argv)
         }
         /* I'm the child, close IO FDs so that ctest continues.  No verbosity here */
         verbose = 0;
-        close(0);
-        close(1);
-        close(2);
+
+        //  Why close a bunch of FDs here?  Because something in the xrootd library is opening new
+        //  fds (upon library init, so before main()) and perhaps using it to track instances?
+        //  Whatever it is doing is interfering with CTest and putting things in the background
+        for (int fd = 0; fd < 13; fd++)
```

Comment:
How do you know how many fds to close? Is there an API in xrootd that can tell you this number?

Reply:
Knowing exactly what was going on (and having it documented) would be nice, but I'd be pretty sure that this is undocumented behaviour.  We're not even calling xrootd subroutines, but apparently library initialization gets you a bunch of open FDs.  This is normally inconsequential, but there is some kind of odd interaction with CTest here and our normal routine of having our remote server fork itself to the background in order to run as a "fixture" in CI doesn't work.  I'd previously observed this with FDs 0,1 and 2, but the Xrootd behaviour seems to expand this list.  We're only doing this in a program that will never use xrootd itself, so this seems safe, albeit weird.  (And I got this number by doing an open() and seeing what was the next FD available.  This could change as xrootd changes, but we'll know because tests that don't use xrootd will start failing.  I guess I could make this configurable, or close the first 128 or something, but lets see how this goes.)